### ISSUE TITLE
Logger introduces a memory issue

### ISSUE THREAD

**Issue by AOx0** [AOx0/proyecto-estructura#37]
Introduced in 08ba4de6342d58941a07533dc938ef20a22bc112

Output:

```
^C    Setting end rwlock to true...
Finishing from rust...
    Setting end rwlock to true...
    Getting runtime...
    Stopping server...
        Joining threads...
            Joining thread...
    Waiting for runtime...
        Done!
    Dropping channels...
    Dropping recv_signal...
Done from rust!
malloc_consolidate(): unaligned fastbin chunk detected
```

**Comment by AOx0** [comment]
Partially fixed by 584444f580b16835ccb56c10bee075a966fa4de8. There seems to be a problem with the isolated scope in the main.cpp, maybe the shared_scope being moved into the inner scope.

```cpp
int main() {
    shared_ptr<Logger> log(make_shared<Logger>(data_path/"log.log"));
    
    // ...
    
    {
      TcpServer server = TcpServer();
  
      while (!st) {
        shared_ptr<optional<Connection>> event = server.recv();
        if (event->has_value()) {
          threads.emplace_back(
              thread([event, &server, log] { resolve(event, server, log); }));
        }
      }
      
      WARN("Shutting down server"); // No errors
    }
    
    // malloc_consolidate(): unaligned fastbin chunk detected
    // WARN("Shutting down server");
    
    
    // ...
}
```

**Comment by AOx0** [comment]
Closing because it's fixed. Maybe I'll reopen if I come back to totally fix it.
